### ISSUE TITLE
Requirements

### ISSUE THREAD

**Issue by simonprovost** [automl/TabPFN#1]
Dear authors,

Wonderful system ! I heard numerous transformers discussions using AutoML but no concrete systems, finally here is one. Interesting though. I am fairly new to AutoML as part of my Ph.D research and not in Transformers anyway but I'll keep an eye on this repository ✅

**Brief of the issue**

Surprisingly, your requirements include installing everything; yet, when I installed TabPFN using pip in my conda environment, the system would not function. I had to re-download (i.e, using PIP) PyTorch and/or AutoGluon for the setup to function. Aside from this brief observation, no console logs were requiring to download them just my instinct as the system was not working.

**To reproduce,** I reckon the following would be worthwhile.

> - Setup a brand-new OSX (I do work on a new OSX but that might also be the case on Windows/Linux);
> - Install Python 3-x and Conda
> - Install TabPFN
> - Run the basic recommended snippet from the README and see whether it does work

This procedure was not functioning on my end; as mentioned, I needed to redownload PyTorch and/or AutoGluon. Perhaps you simply need to resolve this quick issue for further users or indicate in the readme that PyTorch and/or AutoGluon must be downloaded manually.

Hope this help ! Otherwise great contribution,
Cheers,

**Comment by noahho** [comment]
Hi Simon,

Thank you for submitting this issue and documenting so well, really appreciated. I could not replicate your error, however, when I installed our repository from scratch. To be clear:

- you installed TabPFN using pip install tabpfn?
- you ran:

`from sklearn.metrics import accuracy_score
from sklearn.datasets import load_breast_cancer
from sklearn.model_selection import train_test_split

from tabpfn.scripts.transformer_prediction_interface import TabPFNClassifier

X, y = load_breast_cancer(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)

classifier = TabPFNClassifier(device='cpu')
classifier.fit(X_train, y_train)
y_eval, p_eval = classifier.predict(X_test, return_winning_probability=True)

print('Accuracy', accuracy_score(y_test, y_eval))`

- What error did you see?

If you are going to run the entire evaluation pipeline including autogluon, you would need to install autogluon separately (as mentioned in the README).

Good luck with your Ph.D.! if you want to discuss any questions on AutoML + Transformers let us know!

**Comment by simonprovost** [comment]
@noahho There may have been an additional adverse effect. I underwent the exact same procedure. Anyway, let us end this discussion. I apologise for the spam on this one; I will return to it with my own machine later as soon as I got time because now is hectic.  I will test using an empty conda env to determine the outcome later but it is not necessary to keep this discussion open right now.

Assuming it worked on your end, everything is fine I am closing this issue,

Cheers
